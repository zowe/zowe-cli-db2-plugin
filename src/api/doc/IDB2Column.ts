/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Column description
 * @interface IDB2Column
 * @export
 */
export type IDB2Column = {
    /**
     * The catalog name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_CAT: string;

    /**
     * The scheme name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_SCHEM: string;

    /**
     * The table name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_NAME: string;

    /**
     * The column name
     * @type {string}
     * @memberof IDB2Column
     */
    COLUMN_NAME: string;

    /**
     * The numeric representation of the SQL data type
     * @type {number}
     * @memberof IDB2Column
     */
    DATA_TYPE: number;

    /**
     * The string representation of the SQL data type
     * @type {string}
     * @memberof IDB2Column
     */
    TYPE_NAME: string;

    /**
     * The length of the column
     * @type {number}
     * @memberof IDB2Column
     */
    COLUMN_SIZE: number;

    /**
     * Buffer length needed to store data
     * @type {number}
     * @memberof IDB2Column
     */
    BUFFER_LENGTH: number;

    /**
     * Number of decimal digits for a numeric column
     * @type {number}
     * @memberof IDB2Column
     */
    DECIMAL_DIGITS: number | null;

    /**
     * The radix of a numeric column
     * @type {number}
     * @memberof IDB2Column
     */
    NUM_PREC_RADIX: number | null;

    /**
     * Indicates whether the column can contain the null value
     *
     * 0 - NULLs are not allowed
     * 1 - NULLs are allowed
     * @type {number}
     * @memberof IDB2Column
     */
    NULLABLE: number;

    /**
     * Description of the column
     * @type {string}
     * @memberof IDB2Column
     */
    REMARKS: string;

    /**
     * The default value
     * @type {string}
     * @memberof IDB2Column
     */
    COLUMN_DEF: string;

    /**
     * The numeric representation of the SQL data type
     * @type {number}
     * @memberof IDB2Column
     */
    SQL_DATA_TYPE: number;

    /**
     * The subtype of a datetime type
     *
     * 1 - Date
     * 2 - Time
     * 3 - Timestamp
     * @type {number|null}
     * @memberof IDB2Column
     */
    SQL_DATETIME_SUB: number | null;

    /**
     * Length in bytes of the column
     * @type {number|null}
     * @memberof IDB2Column
     */
    CHAR_OCTET_LENGTH: number | null;

    /**
     * Position of the column in the table starting from 1
     * @type {number}
     * @memberof IDB2Column
     */
    ORDINAL_POSITION: number;

    /**
     * Indicates whether the column can contain the null value
     *
     * NO - NULLs are not allowed
     * YES - NULLs are allowed
     * @type {string}
     * @memberof IDB2Column
     */
    IS_NULLABLE: string;
};
