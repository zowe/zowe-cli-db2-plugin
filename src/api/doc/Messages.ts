/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { IMessageDefinition, apiErrorHeader } from "@zowe/imperative";

export const noDB2Input: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 parameters were supplied.",
};

export const noHostName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 server hostname was supplied.",
};

export const noPortNumber: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 server port number was supplied.",
};

export const noUserName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 User ID was supplied.",
};

export const noPassword: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 user password was supplied.",
};

export const noDatabaseName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 database name was supplied.",
};

export const noTableName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 table name was supplied.",
};
