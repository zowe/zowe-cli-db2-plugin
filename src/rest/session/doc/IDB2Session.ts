/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface for DB2 connection parameters
 * @export
 * @interface
 */
export interface IDB2Session {

    /**
     * The name or IP address of the server to connect to
     * @type {string}
     * @memberof IDB2Session
     */
    hostname: string;

    /**
     * The port number where the server is listening
     * @type {number}
     * @memberof IDB2Session
     */
    port: number;

    /**
     * The user ID
     * @type {string}
     * @memberof IDB2Session
     */
    user: string;

    /**
     * The user's password
     * @type {string}
     * @memberof IDB2Session
     */
    password: string;

    /**
     * The database name to use
     * @type {string}
     * @memberof IDB2Session
     */
    database: string;

    /**
     * The path to a SSL Certificate file from a server or a CA signed certificate
     *
     * When specified the connection to the database server will be done over SSL.
     * @type {string}
     * @memberof IDB2Session
     */
    sslFile?: string;
}
