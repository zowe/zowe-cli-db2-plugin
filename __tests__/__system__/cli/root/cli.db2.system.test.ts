/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../__src__/environment/doc/ITestPropertiesSchema";

let TEST_ENV: ITestEnvironment<ITestPropertiesSchema>;

describe("db2 command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENV = await TestEnvironment.setUp({
            installPlugin: true,
            tempProfileTypes: ["db2"],
            testName: "root_command"
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENV);
    });

    it("should display the help", () => {
        const response = runCliScript(__dirname + "/__scripts__/root_help.sh", TEST_ENV);
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
    });

    it("should fail with invalid parameter", async () => {
        const response = runCliScript(__dirname + "/__scripts__/invalid_parameter.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });

    it("should fail with invalid option", async () => {
        const response = runCliScript(__dirname + "/__scripts__/invalid_option.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });
});
