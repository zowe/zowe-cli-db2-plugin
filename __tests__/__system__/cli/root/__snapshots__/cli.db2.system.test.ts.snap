// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`db2 command should display the help 1`] = `
"
 DESCRIPTION
 -----------

   Interact with IBM Db2 for z/OS

 USAGE
 -----

   zowe db2 <group>

   Where <group> is one of the following:

 GROUPS
 ------

   call    Call a stored procedure
   execute Execute a SQL query    
   export  Export a table         

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a group

   --help-web  | --hw (boolean)

      Display HTML help in browser

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: db2.\\",
  \\"stdout\\": \\"\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Interact with IBM Db2 for z/OS\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 <group>\\\\n\\\\n   Where <group> is one of the following:\\\\n\\\\n GROUPS\\\\n ------\\\\n\\\\n   call    Call a stored procedure\\\\n   execute Execute a SQL query    \\\\n   export  Export a table         \\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Interact with IBM Db2 for z/OS\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 <group>\\\\n\\\\n   Where <group> is one of the following:\\\\n\\\\n GROUPS\\\\n ------\\\\n\\\\n   call    Call a stored procedure\\\\n   execute Execute a SQL query    \\\\n   export  Export a table         \\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n\\"
}"
`;

exports[`db2 command should fail with invalid option 1`] = `
"Command Error:
Unknown arguments: command, D
Command failed due to improper syntax
Command entered: \\"zowe db2 --command -D DDF\\"
Use \\"zowe db2 --help\\" to view groups, commands, and options.
Error Details:
Unknown arguments: command, D
"
`;

exports[`db2 command should fail with invalid option 2`] = `""`;

exports[`db2 command should fail with invalid parameter 1`] = `
"Command Error:
Unknown arguments: drop, DEMOUSER.DEMOTABLE
Command failed due to improper syntax
Command entered: \\"zowe db2 drop DEMOUSER.DEMOTABLE\\"
Use \\"zowe db2 --help\\" to view groups, commands, and options.
Error Details:
Unknown arguments: drop, DEMOUSER.DEMOTABLE
"
`;

exports[`db2 command should fail with invalid parameter 2`] = `""`;
